.github/workflows/trigger_builds.yml

```diff
@@ -4,39 +4,47 @@ on:
   push:
     branches-ignore:
       - "renovate/**"
-    # NOTE: `!` doesn't work with `paths-ignore` :(
-    # So we a catch-all glob instead
-    # https://github.com/orgs/community/discussions/25369#discussioncomment-3247674
     paths:
-      - "**"
-      - "!.github/**"
-      - ".github/workflows/build.yml"
-      - ".github/workflows/trigger_builds.yml"
-      - "!flatpak/"
-      - "!nix/"
-      - "!scripts/"
+      # File types
+      - "**.cpp"
+      - "**.h"
+
+      # Directories
+      - "buildconfig/"
+      - "cmake/"
+      - "launcher/"
+      - "libraries/"
+      - "program_info/"
+      - "tests/"
 
-      - "!.git*"
-      - "!.envrc"
-      - "!**.md"
+      # Files
+      - "CMakeLists.txt"
       - "COPYING.md"
-      - "!renovate.json"
-  pull_request:
-    # See above
-    paths:
-      - "**"
-      - "!.github/**"
+
+      # Workflows
       - ".github/workflows/build.yml"
       - ".github/workflows/trigger_builds.yml"
-      - "!flatpak/"
-      - "!nix/"
-      - "!scripts/"
+  pull_request:
+    paths:
+      # File types
+      - "**.cpp"
+      - "**.h"
```

Comment:
Why do we have file types when we already have directories? Does this mean that it will only run when we change a *.cpp/h from that directory, or will it run on all files in the directory, and if extra *.cpp/h files are outside this, would they allow it to be run?

Reply:
The latter
To quote
> as well as any .cpp or
.h files to account for any new folders created - though these should
still be added to the workflow later
Signed-off-by: Seth Flynn [getchoo@tuta.io](mailto:getchoo@tuta.io)

Reply:
Then I would suggest also adding *.java